### PR TITLE
feat: Bitcoin Output descriptors for wallets

### PR DESCRIPTION
**Pull request by markettes**

### CODE REVIEW

File: test/NodeGuard.Tests/Helpers/WalletParserTests.cs [hunk 1]
```diff
@@ -149,4 +150,315 @@ public void ParseOutputDescriptor_InvalidSegwitSingleSig(string descriptor, stri
 
     }
     
+    // Testing GetOutputDescriptor method
+    [Fact]
+    public void GetOutputDescriptor_NativeSegwits()
```
Comment (RodriFS) [inline test/NodeGuard.Tests/Helpers/WalletParserTests.cs:155]: Please if you can order the tests as Arrange/Act/Assert

**Review by RodriFS — DISMISSED** [review]

File: test/NodeGuard.Tests/Helpers/WalletParserTests.cs (hunk 1 above)
Comment (markettes) [reply test/NodeGuard.Tests/Helpers/WalletParserTests.cs:155]: Done! 🏁

File: src/Helpers/WalletParser.cs [hunk 2]
```diff
@@ -105,4 +108,99 @@ DerivationStrategyBase Parse(string str)
             return strategy;
         }
     }
+
+    public static string GetOutputDescriptor(Wallet wallet, string bitcoinNetwork)
```
Comment (Jossec101) [inline src/Helpers/WalletParser.cs:112]: NIT: We could use https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/extension-methods here

File: src/Helpers/WalletParser.cs [hunk 3]
```diff
@@ -105,4 +108,99 @@ DerivationStrategyBase Parse(string str)
...
+    }
+    
+    public static byte[] GetMasterFingerprint(string masterFingerprint)
+    {
+        var internalBytes = Enumerable.Range(0, masterFingerprint.Length)
+            .Where(x => x % 2 == 0)
+            .Select(x => Convert.ToByte(masterFingerprint.Substring(x, 2), 16))
+            .ToArray();
+        return internalBytes;
+    }
```
Comment (Jossec101) [inline src/Helpers/WalletParser.cs:205]: Can you document this function?

**Review by Jossec101 — CHANGES_REQUESTED** [review]
What about checksums? We should enforce them.

File: src/Helpers/WalletParser.cs (hunk 2 above)
Comment (markettes) [reply src/Helpers/WalletParser.cs]: done 🏁

**Review by Jossec101 — APPROVED** [review]

**Pull request merged**